cli/src/pcluster/validators/ec2_validators.py

```diff
@@ -189,27 +192,111 @@ def _validate(self, capacity_reservation_id: str, instance_type: str, subnet: st
                 )
 
 
+def get_capacity_reservations(capacity_reservation_resource_group_arn):
+    capacity_reservation_ids = AWSApi.instance().resource_groups.get_capacity_reservation_ids_from_group_resources(
+        capacity_reservation_resource_group_arn
+    )
+    print(capacity_reservation_ids)
+    return AWSApi.instance().ec2.describe_capacity_reservations(capacity_reservation_ids)
+
+
+def capacity_reservation_matches_instance(capacity_reservation: Dict, instance_type: str, subnet: str) -> bool:
+    return capacity_reservation["InstanceType"] == instance_type and capacity_reservation[
+        "AvailabilityZone"
+    ] == AWSApi.instance().ec2.get_subnet_avail_zone(subnet)
+
+
 class CapacityReservationResourceGroupValidator(Validator):
     """Validate at least one capacity reservation in the resource group can be used with the instance and subnet."""
 
-    def _validate(self, capacity_reservation_resource_group_arn: str, instance_type: str, subnet: str):
+    def _validate(self, capacity_reservation_resource_group_arn: str, instance_types: List[str], subnet: str):
         if capacity_reservation_resource_group_arn:
-            capacity_reservation_ids = (
-                AWSApi.instance().resource_groups.get_capacity_reservation_ids_from_group_resources(
-                    capacity_reservation_resource_group_arn
+            capacity_reservations = get_capacity_reservations(capacity_reservation_resource_group_arn)
+            for instance_type in instance_types:
+                found_qualified_capacity_reservation = False
+                for capacity_reservation in capacity_reservations:
+                    if capacity_reservation_matches_instance(capacity_reservation, instance_type, subnet):
+                        found_qualified_capacity_reservation = True
+                        break
+                if not found_qualified_capacity_reservation:
+                    self._add_failure(
+                        f"Capacity reservation resource group {capacity_reservation_resource_group_arn} must have at "
+                        f"least one capacity reservation for {instance_type} in the same availability zone as subnet"
+                        f" {subnet}.",
```

Comment:
How about adding more information to the message? Something like "... You can suppress the validator if you wish to launch this instance type without a capacity reservation."